shoop/campaigns/models/basket_line_effects.py

```diff
@@ -0,0 +1,74 @@
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import random
+
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from shoop.campaigns.models import BasketDiscountEffect
+from shoop.core.models import OrderLineType, Product
+from shoop.core.order_creator._source import LineSource
+
+
+class BasketLineEffect(BasketDiscountEffect):
```

Comment:
Why is this deriving from BasketDiscountEffect? IMHO it shouldn't, since LineEffect and DiscountEffect are two different things: one creates new lines for basket and another accumulates to the total discount amount.